lib/vfscore/automount.c

```diff
@@ -67,10 +71,103 @@ struct vfscore_volume {
 	unsigned long flags;
 	/* Mount options */
 	const char *opts;
+#if CONFIG_LIBVFSCORE_FSTAB
+	/* Unikraft Mount options, see vfscore_mount_volume() */
+	char *ukopts;
+#endif /* CONFIG_LIBVFSCORE_FSTAB */
 };
 
+#if CONFIG_LIBVFSCORE_FSTAB
+/* Handle `mkpath` Unikraft Mount Option */
+static int vfscore_volume_process_ukopts_do_mkpath(const char *path)
+{
+	const char *curri;
+	char *pos;
+	int rc;
+
+	UK_ASSERT(path);
+
+	curri = path;
+	pos = strchr(path + 1, '/');
+
+	while (pos) {
+		/* If we reached last directory, delete trailing `/` and stop */
+		if (pos[1] == '\0') {
+			*pos = '\0';
+			break;
+		}
+
+		/* Zero out the next '/' */
+		*pos = '\0';
+
+		/* mkdir() with S_IRWXU */
+		rc = mkdir(curri, 0700);
```

Comment:
`curri` is equal to `path`... 🤔   I guess, I am missing something because `curri` would always point to the first cut out substring of `path`. A subdirectory structure would not be correctly created?